### ISSUE TITLE
AnnotationUseStyle: false positive on annotation with default nested annotation inside

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#6540]
Moved from https://github.com/checkstyle/checkstyle/issues/6446#issuecomment-466411459.

````
$ cat TestClass.java
@Anno(@Value) // violation , but no violation is expected
public class Example {
}

@interface Anno {
    Value value();
}

@interface Value {
}

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>

    <module name="TreeWalker">
        <module name="AnnotationUseStyle" />
    </module>
</module>

$ java -jar checkstyle-8.10-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[ERROR] TestClass.java:1: Annotation cannot have closing parenthesis. [AnnotationUseStyle]
Audit done.
Checkstyle ends with 1 errors.
````

Expecting no violation.

**Comment by rnveach** [comment]
Fix was merged
